kate/recovery/src/data.rs

```diff
@@ -34,6 +34,26 @@ impl Cell {
 	}
 }
 
+/// Merges cells data per row.
+/// Cells are sorted before merge.
+pub fn rows(cells: &[Cell]) -> Vec<(u32, Vec<u8>)> {
+	let mut sorted_cells = cells.to_vec();
+	sorted_cells.sort_by(|a, b| {
+		let by_row = a.position.row.cmp(&b.position.row);
+		let by_col = a.position.col.cmp(&b.position.col);
+		by_row.then(by_col)
+	});
```

Comment:
Is not the default implementation if you add `#[derive(PartialOrd, Ord)]` to `Position`? 
(see [Ord Derivable](https://doc.rust-lang.org/std/cmp/trait.Ord.html#derivable):

```suggestion
	sorted_cells.sort_by_key(|cell| cell.position);
```

Reply:
Well it depends whats affects sort more, rows or columns. In this case its row index. Having default implementation would maybe make that implicit?

Reply:
I see, you're right.
What do you about next simplification?
```Rust
sorted_cells.sort_by(|a, b| (a.position.row, a.position.col).cmp(&(b.position.row,b.position.col))
```

Reply:
This is cool, its intuitive to assume that first element of pair has precedence. Ive updated the code.